mirobo/airpurifier.py

```diff
@@ -0,0 +1,185 @@
+from .device import Device
+from typing import Any, Dict
+import enum
+
+
+class OperationMode(enum.Enum):
+    Auto = 'auto'
+    Silent = 'silent'
+    Favorite = 'favorite'
+    Medium = 'medium'
+    High = 'high'
+    Strong = 'strong'
+    Idle = 'idle'
+
+
+class LedBrightness(enum.Enum):
+    Bright = 0
+    Dim = 1
+    Off = 2
+
+
+class AirPurifier(Device):
+    """Main class representing the air purifier."""
+
+    def status(self):
+        """Retrieve properties."""
+
+        # A few more properties:
+        properties = ['power', 'aqi', 'humidity', 'temp_dec',
+                      'mode', 'led', 'led_b', 'buzzer', 'child_lock',
+                      'limit_hum', 'trans_level', 'bright',
+                      'favorite_level', 'filter1_life', 'act_det',
+                      'f1_hour_used', 'use_time', 'motor1_speed']
+
+        values = self.send(
+            "get_prop",
+            properties
+        )
+        return AirPurifierStatus(dict(zip(properties, values)))
+
+    def set_mode(self, mode: OperationMode):
+        """Set mode."""
+
+        # auto, silent, favorite, medium, high, strong, idle
+        return self.send("set_mode", [mode.value])
+
+    def set_favorite_level(self, level: int):
+        """Set favorite level."""
+
+        # Set the favorite level used when the mode is `favorite`,
+        # should be  between 0 and 16.
+        return self.send("favorite_level", [level])  # 0 ... 16
+
+    def set_led_brightness(self, brightness: LedBrightness):
+        """Set led brightness."""
+
+        # bright: 0, dim: 1, off: 2
+        return self.send("set_led_b", [brightness.value])
+
+    def set_led(self, led: bool):
+        """Turn led on/off."""
+        if led:
+            return self.send("set_led", ['on'])
+        else:
+            return self.send("set_led", ['off'])
+
+    def set_buzzer(self, buzzer: bool):
+        """Set buzzer."""
+        if buzzer:
+            return self.send("set_mode", ["on"])
+        else:
+            return self.send("set_mode", ["off"])
+
+    def set_humidity_limit(self, limit: int):
+        """Set humidity limit."""
+
+        # 40, 50, 60, 70 or 80
+        return self.send("set_limit_hum", [limit])
+
+
+class AirPurifierStatus:
+    """Container for status reports from the air purifier."""
+    def __init__(self, data: Dict[str, Any]) -> None:
+        # Response of a Air Purifier Pro:
+        # ['power': 'off', 'aqi': 41, 'humidity': 62, 'temp_dec': 293,
+        #  'mode': 'auto', 'led': 'on', 'led_b': null, 'buzzer': null,
+        #  'child_lock': 'off', 'limit_hum': null, 'trans_level': null,
+        #  'bright': 71, 'favorite_level': 17, 'filter1_life': 77,
+        #  'act_det': null, 'f1_hour_used': 771, 'use_time': 2776200,
```

Comment:
Oh, to add. Those are probably encoded as Nones, trying to construct an enum out of them will cause ValueError, so trying to access `led_brightness` would cause a crash at least. I think 
```if self.data["led_b"] is not None:
    return LedBrightness(self.data["led_b"])
```
is probably the best solution here.